planner/core/physical_plans.go

```diff
@@ -60,6 +60,21 @@ type PhysicalTableReader struct {
 	tablePlan  PhysicalPlan
 }
 
+// GetPhysicalReader returns PhysicalTableReader for logical TableGather.
+func (tg *TableGather) GetPhysicalReader(schema *expression.Schema, stats *property.StatsInfo, props ...*property.PhysicalProperty) *PhysicalTableReader {
+	reader := PhysicalTableReader{}.Init(tg.ctx)
+	reader.stats = stats
+	reader.SetSchema(schema)
+	reader.childrenReqProps = props
+	return reader
+}
+
+// SetChildren overrides PhysicalPlan SetChildren interface.
+func (p *PhysicalTableReader) SetChildren(children ...PhysicalPlan) {
+	p.tablePlan = children[0]
+	p.TablePlans = flattenPushDownPlan(p.tablePlan)
```

Comment:
Do `p.children` also need to be set here?

Reply:
No, `PhysicalTableReader` is the leaf operator in current executor.